### ISSUE TITLE
`date_hierarchy` on related fields in `ForeignKey` doesn't work.

### ISSUE THREAD

**Issue by FraCata00** [fabiocaccamo/django-admin-interface#244]
**Python version**
3.10.6

**Django version**
3.2.16

**Package version**
0.24.2

**Current behavior (bug description)**
if in modeladmin set the date_hierarchy with a field inside the ForeignKey, not work, maybe because not work the template_tags, so i read that inside this, search for the fields of Models, so the fk__fields is not a field of the chilld model

**Expected behavior**
?

**Comment by FraCata00** [comment]
like this 
```python
class ChildModel(admin.ModelAdmin):
          date_hierarchy = "fk__parent_fields"
```
exception like this:
```python
FieldDoesNotExist at /it/app/model/

Model has no field named "fk_model__parent_field"
```

**Comment by fabiocaccamo** [comment]
@FraCata00 thank you for reporting this. 
Yes, I agree with you, the error is in the template tag.

Just for being 100% sure, does it work correctly if you don't use `django-admin-interface`?

**Comment by FraCata00** [comment]
@fabiocaccamo  yes it work 100%

**Comment by FraCata00** [comment]
@fabiocaccamo  if U want, after work I can fix that, I try

**Comment by fabiocaccamo** [comment]
@FraCata00 if you can submit a PR it would be great!

**Comment by FraCata00** [comment]
@fabiocaccamo Great!! Try to fix in this weekend

**Comment by FraCata00** [comment]
> @FraCata00 if you can submit a PR it would be great!

Sorry, how can I start the project? Any docker or docker-compose file doesn't exists

**Comment by merwok** [comment]
See https://docs.djangoproject.com/en/4.1/intro/tutorial01/#creating-a-project and https://github.com/fabiocaccamo/django-admin-interface#installation

**Comment by FraCata00** [comment]
> See https://docs.djangoproject.com/en/4.1/intro/tutorial01/#creating-a-project and https://github.com/fabiocaccamo/django-admin-interface#installation 

Man,I'm a Django developer in a Italy software house😅
I know that but my question is how start a development place only for this app

**Comment by merwok** [comment]
Well you can’t expect people to guess your experience level!

This project doesn’t have a demo app, so you have to write one to test the effect of changes, or rely on unit tests.

**Comment by FraCata00** [comment]
Ohhh okay,really good point 🫶🏼
Thanks for the comment!

**Comment by FraCata00** [comment]
I tried to reproduce this bug, but the bug only appears if a custom theme is active...
with default theme called "Django" the bug isn't reproduced
@fabiocaccamo

**Comment by fabiocaccamo** [comment]
@FraCata00 all themes are handled by the same logic, the only things that can differ are the their options, so I suppose that the theme that causes the problem has `list_filter_removal_links` set, while in the default theme it's that option is not set by default.

**Comment by FraCata00** [comment]
> @FraCata00 all themes are handled by the same logic, the only things that can differ are their options, so I suppose that the theme that causes the problem has `list_filter_removal_links` set, instead in the default theme that option is not set (by default).

I also think that the problem is there, because it triggers that function by setting a custom theme

**Comment by merwok** [comment]
I added that template tag, feel free to ask any question or tag me if you make a PR!

**Comment by FraCata00** [comment]
> I added that template tag, feel free to ask any question or tag me if you make a PR!

Amazing work man, so I'd like to fix that when a custom theme is set

**Comment by FraCata00** [comment]
> This is how the date_hierarchy template tag gets the field object from a string: https://github.com/django/django/blob/main/django/contrib/admin/templatetags/admin_list.py#L356
> 
> 
> 
> The function used there must handle `__` segments, which my simple code `changelist.model._meta.get_field` doesn’t.
> 
> So changing this line should do it!

Exactly,I insert a try except with ```python 
FieldDoesNotExist
```
,and use
```python 
get_field_from_path
``` 
method

**Comment by fabiocaccamo** [comment]
@FraCata00 any update?

**Comment by fabiocaccamo** [comment]
@FraCata00 fixed in `0.25.0` version.

**Comment by FraCata00** [comment]
@fabiocaccamo really good work!
my advice wasn't so wrong :)
